### ISSUE TITLE
DistributedDataParallel has no improvement when using one more machines.

### ISSUE THREAD

**Issue by s974534426** [pytorch/pytorch#63936]
I am using DistributedDataParallel to train my deep learning model. I have two machines. And each of them has four gpus. When I changed to single machine multi-gpu from single machine single gpu, I got a great performance improvement. So I think the code is correct. However, if I changed to two machines multi-gpu from single machine multi-gpu, I got no imporvement. 

What are the possible problems? 

I'm using torch.distributed.launch to start. When I changed to two machines, I just modified the parameters ```nnode``` and ```node_rank```. And I checked the data sampler, I used DistributedSampler. The number of batch is half when I used two machines compared to single machine. 

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu @gcramer23

**Comment by s974534426** [comment]
The code ```loss.backward()``` passes gradient between machines, which spends a lot. So I decreased the size of model and increased the batchsize. Then I solved the problem.
